src/overlays/actors/ovl_Dm_Bal/z_dm_bal.c

```diff
@@ -30,31 +36,185 @@ const ActorInit Dm_Bal_InitVars = {
     (ActorFunc)DmBal_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_060005FC;
-extern UNK_TYPE D_06001804;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/DmBal_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/DmBal_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/func_80C1EAC4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/func_80C1EAD8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/func_80C1EAE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/func_80C1EC60.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/func_80C1ED0C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/func_80C1ED64.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/DmBal_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/func_80C1EF80.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Dm_Bal/func_80C1F060.s")
+static AnimationInfo D_80C1F170[0xE] = {
+    { 0x060005FC, 1.0f, 0.0f, 0.0f, 0, -8.0f }, { 0x06000840, 1.5f, 0.0f, 0.0f, 0, -8.0f },
+    { 0x06000840, 1.5f, 0.0f, 0.0f, 2, -4.0f }, { 0x0600A7DC, 1.0f, 0.0f, 0.0f, 2, -4.0f },
+    { 0x0600B1E8, 1.0f, 0.0f, 0.0f, 0, -8.0f }, { 0x0600B604, 1.0f, 0.0f, 0.0f, 0, -8.0f },
+    { 0x0600C498, 1.0f, 0.0f, 0.0f, 2, -8.0f }, { 0x0600C8D8, 1.0f, 0.0f, 0.0f, 0, -8.0f },
+    { 0x0600C8D8, 1.0f, 0.0f, 0.0f, 2, -8.0f }, { 0x0600C498, 1.0f, 23.0f, 0.0f, 2, -8.0f },
+    { 0x0600D530, 1.0f, 0.0f, 0.0f, 0, -8.0f }, { 0x06000C78, 1.0f, 0.0f, 0.0f, 0, -2.0f },
+    { 0x0600CB78, 1.0f, 0.0f, 0.0f, 0, -2.0f }, { 0x06001804, 1.0f, 0.0f, 0.0f, 0, -2.0f },
+};
```

Comment:
These `0x06XXXXXX` addresses are segment pointers that come from the actors object. Could you change them to use the correct symbols from there. Also the second to last argument (the integer) should use `AnimationModes` enum found in `z64animation.h`. In this case `0 -> ANIMMODE_LOOP` and `2 -> ANIMMODE_ONCE`. Last we usually leave the array length of array declarations, i.e. arrays that have there data declared with them. In this case that would just mean removing the `0xE`.

Reply:
done